Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -120,11 +122,59 @@ StyledMarkedText::Style StyledMarkedText::computeStyleForUnmarkedMarkedText(cons
     return style;
 }
 
+static Vector<StyledMarkedText> coalesceAdjacentWithSameRanges(const Vector<StyledMarkedText>& styledTexts)
+{
+    Vector<StyledMarkedText> frontmostMarkedTexts;
+    frontmostMarkedTexts.append(styledTexts[0]);
+    for (auto it = styledTexts.begin() + 1, end = styledTexts.end(); it != end; ++it) {
+        StyledMarkedText& previousStyledMarkedText = frontmostMarkedTexts.last();
+        if (previousStyledMarkedText.startOffset == it->startOffset && previousStyledMarkedText.endOffset == it->endOffset) {
+            // Marked texts completely cover each other
+            previousStyledMarkedText.style.backgroundColor = blendSourceOver(previousStyledMarkedText.style.backgroundColor, it->style.backgroundColor);
+            // Assign priority to higher priority highlight
+            previousStyledMarkedText.priority = it->priority;
+            if (!it->highlightName.isNull() && !previousStyledMarkedText.highlightName.isNull())
+                previousStyledMarkedText.highlightName = it->highlightName;
+            if (it->style.hasExplicitStrokeColor)
+                previousStyledMarkedText.style.textStyles.fillColor = it->style.textStyles.fillColor;
+            continue;
+        }
+        frontmostMarkedTexts.append(*it);
+    }
+    return frontmostMarkedTexts;
+}
+
+static void orderHighlights(const ListHashSet<AtomString>& markedTextsNames, Vector<MarkedText>& markedTexts)
+{
+    // Maintain copy of old unordered markedTexts
+    Vector<MarkedText> unorderedMarkedTexts = markedTexts;
+    markedTexts.clear();
+    // Append all highlights first, in order
+    for (auto& highlightName : markedTextsNames) {
+        for (auto& element : unorderedMarkedTexts) {
+            if (element.highlightName == highlightName)
+                markedTexts.append(element);
+        }
+    }
+    // Append selection last, to maintain order
+    for (auto& element : unorderedMarkedTexts) {
+        if (element.highlightName.isNull())
```

Comment:
Same as above, not sure why this is an element, and I'm not sure why you're checking if it's got a null highlight name?

Reply:
When the markedText does not have a highlightName, it is usually an indication that it is a selection markedText or essentially not a custom highlight.